hutch_python/epics_arch.py

```diff
@@ -156,33 +156,93 @@ def create_arch_file(experiment, level=None, hutch=None, path=None, dry_run=Fals
+    Gather all user obejcts from the CDS tab in the questionnaire.
+    Parse objects and sperate them based on type.
+    Display them in the console vie PrettyTable.
 
     Parameters
     ----------
     exp: ``str``
         The experiment's name e.g. xppx1003221
     run: ''str''
         The run number e.g. run21
-
+    ----------
     Outputs
     -------
         PrettyTable visualization of cds objects
+    -------
+
     """
-    # pull run data from questionnaire api, then take the data and sort it
-    # create PrettyTable instance and if the values from the run data contain
-    # pcdssetup then put them into a seperate dictionary as they are cds items
-    logger.debug('pull_cds_items(%s)', exp)
-    client = QuestionnaireClient()
-    logger.debug("in cds items, run numb:", str(run[1]))
-    runDetails_Dict = client.getProposalDetailsForRun(str(run[0]), str(run[1]))
+    """
+    pull run data from questionnaire api, then take the data and sort it
+    create Pretty Table instance and if the values from the run data contain pcdssetup
+    then put them into a seperate dictionary as they are cds items
+    """
+
+    logger.debug('pull_cds_items:', exp)
+    try:
+        client = QuestionnaireClient()
+    except GSSError:
+        print('No kerberos token found; rerun after running kinit')
+        sys.exit()
+
+    formatted_run_id = ''
+    run_num = ''
+
+    # handle formatting of proposal id, want: e.g. X-10032
+    # Case -  Expected Format: e.g. xppx1003221
+    if re.match('^.{4}[0-9]{7}$', exp):
+        logger.debug("experiment name format")
+        run_num = 'run'+exp[-2:]
+        logger.debug('run num', run_num)
+        run_id = str(exp[3:-2])
+        logger.debug('run_id', run_id)
+        formatted_run_id = run_id.capitalize()
+        formatted_run_id = formatted_run_id[:1] + '-' + formatted_run_id[1:]
+
+    elif re.match('[A-Z]{1}-[0-9]{5}$', exp) or re.match('^[A-Z]{2}[0-9]{2}$', exp):
+        # Case - Proposal ID Format, have user enter run num manually
+        logger.debug("run_id format")
+        formatted_run_id = exp
+        run_num = 'run' + str(input('Please enter run number: '))
+    else:
+        print('Unrecognized format, please follow the prompts to find experiment data.')
+
+        run_num = 'run' + input('Please enter run number: ')
+        formatted_run_id = input('Please enter proposal ID: ')
+
+    logger.debug('formatted run_id', formatted_run_id)
+    logger.debug('run_num', run_num)
+
+    matchedKey = ''
+
+    try:
+        runDetails_Dict = client.getProposalsListForRun(run_num)
+        for key, vals in runDetails_Dict.items():
+            logger.debug(formatted_run_id, key)
+            if str(key) == str(formatted_run_id):
+                logger.debug('matched key', key)
+                matchedKey = key
+
+    except Exception as e:
+        print("An invalid https request, please check the run number, proposal id and experiment number:", e)
+        sys.exit()
```

Comment:
Consider again logging exceptions and returning or raising instead of exiting the entire program
```suggestion
        logger.error("An invalid https request, please check the run number, proposal id and experiment number:", e)
        return
```